Services/UI/OJS.Services.Ui.Models/Submissions/SubmissionDetailsServiceModel.cs

```diff
@@ -60,42 +53,68 @@ public class SubmissionDetailsServiceModel : IMapExplicitly
 
         public void RegisterMappings(IProfileExpression configuration)
-            => configuration.CreateMap<Submission, SubmissionDetailsServiceModel>()
-                .ForMember(s => s.User, opt => opt.MapFrom(s => s.Participant.User))
-                .ForMember(d => d.MaxUsedMemory, opt => opt.MapFrom(source =>
-                    source.TestRuns.Any()
-                        ? source.TestRuns.Max(tr => tr.MemoryUsed)
-                        : 0.0))
-                .ForMember(d => d.MaxUsedTime, opt => opt.MapFrom(source =>
-                    source.TestRuns.Any()
-                        ? source.TestRuns.Max(tr => tr.TimeUsed)
-                        : 0.0))
+            => configuration
+                .CreateMap<Submission, SubmissionDetailsServiceModel>()
+                .ForMember(d => d.MaxUsedMemory, opt => opt.MapFrom(s =>
+                    GetMaxMemoryAndTimeUsed(s.TestRunsCache).MaxMemoryUsed))
+                .ForMember(d => d.MaxUsedTime, opt => opt.MapFrom(s =>
+                    GetMaxMemoryAndTimeUsed(s.TestRunsCache).MaxTimeUsed))
                 .ForMember(d => d.Content, opt => opt.MapFrom(s =>
                     s.IsBinaryFile
                         ? null
                         : s.ContentAsString))
                 .ForMember(d => d.IsOfficial, opt => opt.MapFrom(s => s.Participant.IsOfficial))
-                .ForMember(d => d.ByteContent, opt => opt.MapFrom(s => s.Content))
+                .ForMember(d => d.User, opt => opt.MapFrom(s => s.Participant.User))
                 .ForMember(s => s.IsProcessed, opt => opt.MapFrom(s => s.Processed))
-                .ForMember(d => d.Tests, opt => opt.MapFrom(s => s.Problem.Tests))
-                .ForMember(d => d.ContestId, opt => opt.MapFrom(s => s.Problem.ProblemGroup.ContestId))
-                .ForMember(d => d.ContestName, opt => opt.MapFrom(s => s.Problem.ProblemGroup.Contest.Name))
-                .ForMember(d => d.ContestCategoryId, opt => opt.MapFrom(s => s.Problem.ProblemGroup.Contest.CategoryId))
+                .ForMember(d => d.ContestId, opt => opt.MapFrom(s => s.Participant.ContestId))
+                .ForMember(d => d.ContestName, opt => opt.MapFrom(s => s.Participant.Contest.Name))
+                .ForMember(d => d.ContestCategoryId, opt => opt.MapFrom(s => s.Participant.Contest.CategoryId))
                 .ForMember(d => d.MaxPoints, opt => opt.MapFrom(s => s.Problem.MaximumPoints))
                 .ForMember(d => d.TotalTests, opt => opt.Ignore())
                 .ForMember(s => s.UserIsInRoleForContest, opt => opt.Ignore())
-                .ForMember(s => s.IsEligibleForRetest, opt => opt.Ignore());
+                .ForMember(s => s.IsEligibleForRetest, opt => opt.Ignore())
+                .ForMember(s => s.Tests, opt => opt.Ignore())
+                .ForMember(s => s.TestRuns, opt => opt.Ignore());
+
+        private static (long? MaxMemoryUsed, int? MaxTimeUsed) GetMaxMemoryAndTimeUsed(string? testRunsCache)
```

Comment:
Extract this into common method and reuse it in all models, where needed.

Reply:
Done.